### PR TITLE
Cameron add finals info service v2

### PR DESCRIPTION
**Pull request by cameronbyrne4**
NOTE: this is an updated branch of my previous code that won't have heinous merge conflicts, and thus should work better.

Completes: https://github.com/ucsb-cs156-f24/proj-courses-f24-01/issues/30
Base reference: https://github.com/ucsb-cs156-f24/proj-courses-f24-01/issues/6
Dokku Deployment: https://courses-dev-cameronbyrne4.dokku-01.cs.ucsb.edu/

Added back end for finals information. Added a method that finals info from an endpoint in the UCSB API, with tests. Also added endpoint where a given quarter (YYYYQ) and enrollment code will get the finals info.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/courses/controllers/ApiController.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
+// testing how creating a draft pull request works! cameron
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/ApiController.java]: Can we remove this from the PR?

**Review by pconrad — CHANGES_REQUESTED** [review]
See code review

**Review by vincentc0202 — COMMENTED** [review]
Delete the commented out test

**Comment by cameronbyrne4** [comment]
Now deployed. Please review and add admin review label!

File: src/main/java/edu/ucsb/cs156/courses/controllers/ApiController.java (hunk 1 above)
Comment (cameronbyrne4) [reply src/main/java/edu/ucsb/cs156/courses/controllers/ApiController.java]: done

**Review by vincentc0202 — DISMISSED** [review]
Looks good, everything looks synchronous

**Review by cameronbyrne4 — COMMENTED** [review]
Pretty sure I deleted in a recent commit

**Review by raysujf — COMMENTED** [review]
I got an error in testing the finalinfo api implementation.
Screen Shot 2024-11-26 at 5 23 51 PM
Screen Shot 2024-11-26 at 5 24 11 PM

**Review by pconrad — DISMISSED** [review]
Code still looks good... will try again to test on dokku

**Comment by raychen179** [comment]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**